pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,830 @@
+                                   disk_thickness=1.0e-3, apex_distance=30e-6)
+
+    Parameters
+    ----------
+    energy : number
+        Beam Energy
+    radius : `float`
+    material : `str`
+        Chemical formula. Default = 'Be'
+    density : `float`
+        Material density in g/cm^3
+    fwhm_unfocused : `float`
+        This is about 400 microns at XPP.
+    disk_thickness : `float`
+        Default = 1.0e-3
+    apex_distance : `float`
+        Default = 30e-6
+
+    Returns
+    -------
+    transmission : `float`
+        Transmission for a single lens
+    '''
+    # mu = mass attenuation coefficient?
+    mu = 1.0 / get_att_len(energy, material=material, density=None)
+
+    sigma = gaussian_fwhm_to_sigma(fwhm_unfocused)
+    # TODO: what is S - Responsivity of a lens?
+    S = (sigma ** (-2.0) + 2.0 * mu / radius) ** (-0.5)
+    aperture_radius = calc_lens_aperture_radius(radius=radius,
+                                                disk_thickness=disk_thickness,
+                                                apex_distance=apex_distance)
+
+    transmission = ((S ** 2 / sigma ** 2) * np.exp(-mu * apex_distance) *
+                    (1 - np.exp(-(aperture_radius ** 2.0) / (2.0 * S ** 2))))
+    return transmission
+
+
+def calc_trans_lens_set(energy, lens_set, material="Be", density=None,
+                        fwhm_unfocused=900e-6, disk_thickness=1.0e-3,
+                        apex_distance=30e-6):
+    '''
+    Calculate  the transmission of a lens set.
+    These would allow us to estimate the total transmission of the lenses
+
+    TODO: where is this document is this message below still relevant?
+    There is latex document that explains the formula.
+    Can be adapted to use different thicknesses for each lens,
+    and different apex distances, but this would require changing
+    the format of lens_set, which would mean changing
+    a whole bunch of other programs too.
+
+    Usage:
+
+    .. code-block:: python
+
+        calc_trans_lens_set(energy, lens_set, material="Be", density=None,
+                            fwhm_unfocused=900e-6)
+
+    Parameters
+    ----------
+    energy : number
+        Beam Energy
+    lens_set : `list`
+        [numer1, lensthick1, number2, lensthick2...]
+    material : `str`
+        Chemical formula. Default = 'Be'
+    density : `float`
+        Material density in g/cm^3
+    fwhm_unfocused : `float`
+        This is about 400 microns at XPP. Default = 900e-6
+    disk_thickness : `float`
+        Default = 1.0e-3
+    apex_distance : `float`
+        Default = 30e-6
+
+    Returns
+    -------
+    transmission : `float`
+        Transmission for a set of lens
+    '''
+
+    apex_distance_tot = 0
+    radius_total_inv = 0
+    # TODO: how can i fix this hack here:
+    # this is an ugly hack: the radius will never be bigger than 1m,
+    # so will always be overwritten
+    radius_aperture = 1.0
```

Comment:
Not so ugly a hack as the above comment indicates. "Better" option might be to build and work with well-named lists (e.g., `apertures = [... for num, radius in lens_set]; radius_aperture = min(apertures)`)

Reply:
Ok I want to make sure I understand this one. I would have something like this: ??
```
 if isinstance(lens_set, int):
        lens_set = get_lens_set(lens_set)
    apertures = list(lens_set[1::2])
    radius_aperture = min(apertures)
```
  don't I need `radius_aperture = max(apertures)`?

I might not understand exactly what this is doing....

Reply:
It's hard to further explain without just showing it in code:

```python
lens_set = list(zip(lens_set[::2], lens_set[1::2]))
radius_total_inv = sum(num / radius for num, radius in lens_set)
apex_distance_tot = sum(num * apex_distance for num, _ in lens_set)
radius_aperture = min(np.sqrt(radius * (disk_thickness - apex_distance))
                      for _, radius in lens_set)
```

(It's very possible I messed up something above, so don't take it verbatim - it's just an example of how to remove the for-loop and arbitrary starting value for `min(...)`)

Reply:
ohhh got it, thanks so much - sorry I did not realize that we're trying to do. 🤦 
I was trying to:
```
radius_aperture = max(apertures)
```
and then in the loop again: 
```
radius_aperture = min(radius_aperture, new_rad_ap)
```

Thanks a lot!!

Reply:
Full disclosure: I didn't look at the original code, just what's in the PR. Please do double/triple check it.

Reply:
I changed this to use your suggested code in 33e8f54, and passed the tests, I am getting the same results. 
I am pretty convinced that it works the same in the old code, I implemented these changes in the old code too and get the same results. 
Thanks a lot

Reply:
Great 👍